File: VanityRoles-Bot-main/VanityRoles-Bot-main/logger.py
"""
: ! Aegis !
    + Discord: itsfizys
    + Community: https://discord.gg/aerox (AeroX Development )
    + for any queries reach out Community or DM me.
"""

import sys
import os
import traceback
from datetime import datetime

try:
    import pytz
    _tz = pytz.timezone("Asia/Kolkata")
except Exception:
    _tz = None

_COLORS = {
    "info":    ("#2F6FD6", "#E5E9F0"),
    "success": ("#0FA37F", "#E5E9F0"),
    "warn":    ("#C47A00", "#E5E9F0"),
    "error":   ("#C2362B", "#E5E9F0"),
    "debug":   ("#6B6B6B", "#E5E9F0"),
}

_MSG_COLOR  = "#D8DEE9"
_TIME_COLOR = "#7A7A7A"


def _rgb(hex_color):
    h = hex_color.lstrip("#")
    r, g, b = int(h[0:2], 16), int(h[2:4], 16), int(h[4:6], 16)
    return lambda t: f"\x1b[38;2;{r};{g};{b}m{t}\x1b[0m"


def _badge(bg_hex, fg_hex):
    bh = bg_hex.lstrip("#")
    fh = fg_hex.lstrip("#")
    br, bgc, bb = int(bh[0:2], 16), int(bh[2:4], 16), int(bh[4:6], 16)
    fr, fg, fb = int(fh[0:2], 16), int(fh[2:4], 16), int(fh[4:6], 16)
    return lambda t: f"\x1b[48;2;{br};{bgc};{bb}m\x1b[38;2;{fr};{fg};{fb}m {t} \x1b[0m"


_msg_paint  = _rgb(_MSG_COLOR)
_time_paint = _rgb(_TIME_COLOR)

_badges = {k: _badge(bg, fg) for k, (bg, fg) in _COLORS.items()}


def _now():
    if _tz:
        return datetime.now(_tz).strftime("%H:%M:%S")
    return datetime.utcnow().strftime("%H:%M:%S")


_LOG_DIR  = os.path.join(os.path.dirname(os.path.abspath(__file__)), "logs")
_LOG_FILE = os.path.join(_LOG_DIR, "bot.log")
os.makedirs(_LOG_DIR, exist_ok=True)


def _write_file(level, context, msg, error=None):
    ts = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
    with open(_LOG_FILE, "a", encoding="utf-8") as f:
        f.write(f"[{ts}] [{level.upper()}] [{context}] {msg}\n")
        if error:
            traceback.print_exception(type(error), error, error.__traceback__, file=f)


def _log(level, context, msg, error=None):
    _write_file(level, context, msg, error)
    if level not in ("info", "success", "warn", "error"):
        return
    line = (
        f"{_time_paint(_now())} "
        f"{_badges[level](context):} "
        f"{_msg_paint(msg)}"
    )
    sys.stdout.write(line + "\n")
    sys.stdout.flush()


class _Logger:
    def info(self, context: str, msg: str):
        _log("info", context, msg)

    def success(self, context: str, msg: str):
        _log("success", context, msg)

    def warn(self, context: str, msg: str):
        _log("warn", context, msg)

    def error(self, context: str, msg: str, exc: Exception | None = None):
        _log("error", context, msg, exc)

    def debug(self, context: str, msg: str):
        _log("debug", context, msg)

    def banner(self):
        cyan  = _rgb("#4FC3F7")
        blue  = _rgb("#2F6FD6")
        dim   = _rgb("#4C566A")
        white = _rgb("#ECEFF4")
        green = _rgb("#0FA37F")

        lines = [
            "",
            cyan  ("  ██████╗  ███████╗██████╗  ██████╗ ██╗  ██╗"),
            cyan  ("  ██╔══██╗██╔════╝██╔══██╗██╔═══██╗╚██╗██╔╝"),
            blue  ("  ███████║█████╗  ██████╔╝██║   ██║ ╚███╔╝ "),
            blue  ("  ██╔══██║██╔══╝  ██╔══██╗██║   ██║ ██╔██╗ "),
            white ("  ██║  ██║███████╗██║  ██║╚██████╔╝██╔╝ ██╗"),
            white ("  ╚═╝  ╚═╝╚══════╝╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝"),
            "",
            dim   ("  ─────────────────────────────────────────────────"),
            f"  {green('●')} {white('GuildTag Bot')}   {dim('|')}   {white('discord.gg/aerox')}",
            f"  {green('●')} {white('Developer')}      {dim('|')}   {white('itsfizys · AeroX Development')}",
            dim   ("  ─────────────────────────────────────────────────"),
            "",
        ]
        for line in lines:
            sys.stdout.write(line + "\n")
        sys.stdout.flush()


logger = _Logger()

File: Appy-Bot-main/Appy-Bot-main/src/lib/logger.js
import fs   from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

function parseHex(hex) {
    const h = hex.replace('#', '');
    return [parseInt(h.slice(0, 2), 16), parseInt(h.slice(2, 4), 16), parseInt(h.slice(4, 6), 16)];
}

const rgb = (hex) => {
    const [r, g, b] = parseHex(hex);
    return (t) => `\x1b[38;2;${r};${g};${b}m${t}\x1b[0m`;
};

const badge = (bgHex, fgHex) => {
    const [br, bg, bb] = parseHex(bgHex);
    const [fr, fg, fb] = parseHex(fgHex);
    return (t) => `\x1b[48;2;${br};${bg};${bb}m\x1b[38;2;${fr};${fg};${fb}m ${t} \x1b[0m`;
};

const LEVEL_BADGES = {
    info:    badge('#2F6FD6', '#E5E9F0'),
    success: badge('#0FA37F', '#E5E9F0'),
    warn:    badge('#C47A00', '#E5E9F0'),
    error:   badge('#C2362B', '#E5E9F0'),
};

const paintMsg  = rgb('#D8DEE9');
const paintTime = rgb('#7A7A7A');

const LOG_DIR  = path.join(__dirname, '../../logs');
const LOG_FILE = path.join(LOG_DIR, 'bot.log');
try { fs.mkdirSync(LOG_DIR, { recursive: true }); } catch {}

function writeFile(level, context, msg) {
    const ts = new Date().toISOString().replace('T', ' ').slice(0, 19);
    try { fs.appendFileSync(LOG_FILE, `[${ts}] [${level.toUpperCase()}] [${context}] ${msg}\n`); } catch {}
}

function now() {
    return new Date().toLocaleTimeString('en-GB', { hour12: false });
}

function log(level, context, msg) {
    writeFile(level, context, msg);
    process.stdout.write(
        `  ${paintTime(now())}  ${LEVEL_BADGES[level](context)}  ${paintMsg(msg)}\n`
    );
}

function banner() {
    const cyan   = rgb('#7DD3FC');
    const indigo = rgb('#818CF8');
    const white  = rgb('#F8FAFC');
    const dim    = rgb('#4C566A');
    const green  = rgb('#0FA37F');
    const soft   = rgb('#E2E8F0');

    const lines = [
        '',
        cyan  ('   █████╗ ███████╗██████╗  ██████╗ ██╗  ██╗'),
        cyan  ('  ██╔══██╗██╔════╝██╔══██╗██╔═══██╗╚██╗██╔╝'),
        indigo('  ███████║█████╗  ██████╔╝██║   ██║ ╚███╔╝ '),
        indigo('  ██╔══██║██╔══╝  ██╔══██╗██║   ██║ ██╔██╗ '),
        white ('  ██║  ██║███████╗██║  ██║╚██████╔╝██╔╝ ██╗'),
        white ('  ╚═╝  ╚═╝╚══════╝╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝'),
        '',
        dim   ('  ─────────────────────────────────────────────────'),
        `  ${green('●')}  ${soft('Bot')}          ${dim('│')}   ${soft('Applications Bot')}`,
        `  ${green('●')}  ${soft('Developer')}    ${dim('│')}   ${soft('Ayl')}`,
        `  ${green('●')}  ${soft('Support')}      ${dim('│')}   ${soft('https://discord.gg/aerox')}`,
        dim   ('  ─────────────────────────────────────────────────'),
        '',
    ];
    process.stdout.write(lines.join('\n') + '\n');
}

export const logger = {
    info:    (context, msg) => log('info',    context, msg),
    success: (context, msg) => log('success', context, msg),
    warn:    (context, msg) => log('warn',    context, msg),
    error:   (context, msg) => log('error',   context, msg),
    banner,
};

/**
 * Project: Applications Bot
 * Author: aliyie (Ayl)
 * Organization: AeroX Development
 * GitHub: https://github.com/AeroXDevs
 * License: Custom
 *
 * © 2026 AeroX Development. All rights reserved.
 */
